BSB_LAN/BSB_LAN.ino

```diff
@@ -6564,49 +6566,62 @@ void loop() {
 
 
 #ifdef LOGGER
-#if defined(ESP32)
-  uint64_t freespace = 0;
-  freespace = SD.totalBytes() - SD.usedBytes();
-#else
-  uint32_t freespace = 0;
-  freespace = SD.vol()->freeClusterCount();
-#endif
-  if ((LoggingMode & CF_LOGMODE_SD_CARD) && freespace > MINIMUM_FREE_SPACE_ON_SD) {
+  if ((LoggingMode & (CF_LOGMODE_SD_CARD|CF_LOGMODE_UDP))) {
     if (((millis() - lastLogTime >= (log_interval * 1000)) && log_interval > 0) || log_now > 0) {
-//    SetDateTime(); // receive inital date/time from heating system
       log_now = 0;
-      File dataFile = SD.open(datalogFileName, FILE_APPEND);
-
-      if (dataFile) {
-        for (int i=0; i < numLogValues; i++) {
-          int outBufLen = 0;
-          if (log_parameters[i] > 0) {
-            outBufLen += sprintf_P(outBuf + outBufLen, PSTR("%lu;%s;%d;"), millis(), GetDateTime(outBuf + outBufLen + 80), log_parameters[i]);
-#ifdef AVERAGES
+      File dataFile;
+      if (LoggingMode & CF_LOGMODE_SD_CARD) {
+        #if defined(ESP32)
```

Comment:
We have decided a while ago that Indention of preprocessor directives should be independent of code indention (and yes, this has not been changed in the whole codebase). Please adjust the indention for the preprocessor directives in the parts of the code that you submitted.

Reply:
Sorry, I wasn't aware of a coding style guideline for BSB_LAN, so I formatted the code section I worked on to make it readable to me.

Btw, consistent formatting could easily achieved by using a pretty printer on the source files. Back in the previous millenium, when I was more active in programming, I used indent, which seems to be still around (https://man.openbsd.org/indent.1). Personally, I would do that for "production" code, as well as using a code checker that is stricter than most compilers (e.g. https://docs.oracle.com/cd/E19205-01/820-4180/man1/lint.1.html), and follow its suggestions.

Reply:
P.S. However, using a pretty printer on all source files would most likely destroy the "clever" indentation in my /DG implementation in html_strings.h

Reply:
Yes, and that's why we have dismissed using pretty printers after similar discussions on other issues before.

Reply:
P.P.S. But at least with the venerable "indent" pretty printer, preserving "clever" formatting is easily done: Just embed those code sections in "/* *INDENT-OFF* */ ... /* *INDENT-ON* */". (The last time I used indent myself was in the previous millenium, so I had to look that one up.)

Personally, I would find a common coding style for projects with multiple contributors helpful, especially since it could be easily/automatically enforced. Btw, of all the coding style descriptions I've seen so far, the one I like best is https://www.kernel.org/doc/html/v4.10/process/coding-style.html - although I've clearly not been following it here.
As it is, BSB_LAN code would not get highest marks from me for coding style / maintainability - but neither does my own work. ;)

Reply:
Well, I don't care for other people's marks on coding style, so I guess that's that ;)...